lib/python/picongpu/utils/picongpu_memory_calculator.py

```diff
@@ -0,0 +1,363 @@
+            target_n_z = self.n_z
+
+        # memory required by the standard particle attributes
+        standard_attribute_mem = np.array([
+            3 * 4,  # momentum
+            3 * 4,  # position
+            1 * 8,  # multimask
+            1 * 8,  # cell index in supercell (``lcellId_t``)
+            1 * 8  # weighting
+        ])
+
+        # memory per particle for additional attributes {unit: byte}
+        additional_mem = num_additional_attributes * 4
+        # \TODO we assume 4 bytes here - check if that's really the case
+        # cells filled by the target species
+        local_cells = target_n_x * target_n_y * target_n_z
+
+        req_mem = local_cells * (np.sum(
+            standard_attribute_mem) + additional_mem) * particles_per_cell
+        return req_mem
+
+    def mem_req_by_rng(
+            self,
+            n_x=None,
+            n_y=None,
+            n_z=None,
+    ):
+        """
+        Memory reserved for the random number generator state on each GPU.
+        The RNG we use is: MRG32ka
+
+        Parameters
+        ----------
+        n_x : int
+            number of cells in x direction (per accelerator)
+        n_y : int
+            number of cells in y direction (per accelerator)
+        n_z : int
+            number of cells in z direction (per accelerator)
+
+        Returns
+        -------
+
+        req_mem : int
+            required memory {unit: bytes} per accelerator
+        """
+        if n_x is None:
+            n_x = self.n_x
+        if n_y is None:
+            n_y = self.n_y
+        if n_z is None:
+            n_z = self.n_z
+
+        req_mem_per_cell = 6 * 8  # bytes
+        local_cells = n_x * n_y * n_z
+
+        req_mem = req_mem_per_cell * local_cells
+        return req_mem
+
+    def mem_req_by_calorimeter(
+            self,
+            n_energy,
+            n_yaw,
+            n_pitch,
+            value_size=None
+    ):
+        """
+        Memory required by the particle calorimeter plugin.
+        Each of the (``n_energy`` x ``n_yaw`` x ``n_pitch``) bins requires
+        a value (32/64 bits). The whole calorimeter is represented twice on
+        each accelerator, once for particles in the simulation and once
+        for the particles that leave the box.
+
+        Parameters
+        ----------
+
+        n_energy : int
+            number of bins on the energy axis
+        n_yaw : int
+            number of bins for the yaw angle
+        n_pitch : int
+            number of bins for the pitch angle
+        value_size : int
+            value size in particle calorimeter {unit: byte} (default: 4)
+
+        Returns
+        -------
+
+        req_mem : int
+            required memory {unit: bytes} per accelerator
+        """
+        if value_size is None:
+            value_size = self.value_size
+
+        req_mem_per_bin = value_size
+        num_bins = n_energy * n_yaw * n_pitch
+        req_mem = req_mem_per_bin * num_bins * 2
```

Comment:
please add a comment for this `2`- you wrote the description in the doc-string above, but please repeat it here, so that this is directly clear